### PR TITLE
Remove useless clippy hints and fix some issues following clippy lints

### PR DESCRIPTION
**Pull request by sticnarf**
Signed-off-by: Yilin Chen <sticnarf@gmail.com>

## What have you changed? (mandatory)

I notice some clippy lints are actually useless. They can be removed now.

And we'd better follow clippy if there is no special reason. I also fix some of them.

## What are the type of changes? (mandatory)

- Engineering (engineering change which doesn't change any feature or fix any issue)

## How has this PR been tested? (mandatory)

CI

## Does this PR affect documentation (docs) or release note? (mandatory)

No

## Does this PR affect `tidb-ansible` update? (mandatory)

No

### CODE REVIEW

**Comment by siddontang** [comment]
@sticnarf CI failed

File: components/codec/src/byte.rs [hunk 1]
```diff
@@ -918,10 +917,9 @@ mod tests {
                     MemComparableByteCodec::encode_all(src, dest);
                 }
             }
-            #[allow(clippy::needless_range_loop)]
-            for i in encoded_prefix_len + encoded_len..encoded_suffix_len {
-                payload_encoded[i] = rand::random();
-            }
+            payload_encoded[encoded_prefix_len + encoded_len..encoded_suffix_len]
```
Comment (sticnarf) [inline components/codec/src/byte.rs]: This line panics when `encoded_suffix_len < encoded_prefix_len + encoded_len`. The original code doesn't panic but it does look strange. Should the range here be `encoded_prefix_len + encoded_len..`?  @breeswish

File: components/codec/src/byte.rs [hunk 2]
```diff
@@ -905,10 +905,9 @@ mod tests {
             let encoded_len = MemComparableByteCodec::encoded_len(payload_len);
             let mut payload_encoded: Vec<u8> =
                 vec![0; encoded_prefix_len + encoded_len + encoded_suffix_len];
-            #[allow(clippy::needless_range_loop)]
-            for i in 0..encoded_prefix_len {
-                payload_encoded[i] = rand::random();
-            }
+            payload_encoded[0..encoded_prefix_len]
+                .iter_mut()
+                .for_each(|byte| *byte = rand::random());
```
Comment (kennytm) [inline components/codec/src/byte.rs]: 😨 

Please use [`fill_bytes`](https://docs.rs/rand/0.7.0/rand/trait.RngCore.html#tymethod.fill_bytes).

```rust
let mut rng = rand::thread_rng();
rng.fill_bytes(&mut payload_encoded[..encoded_prefix_len]);
```

File: components/codec/src/byte.rs [hunk 3]
```diff
@@ -918,10 +917,9 @@ mod tests {
                     MemComparableByteCodec::encode_all(src, dest);
                 }
             }
-            #[allow(clippy::needless_range_loop)]
-            for i in encoded_prefix_len + encoded_len..encoded_suffix_len {
-                payload_encoded[i] = rand::random();
-            }
+            payload_encoded[encoded_prefix_len + encoded_len..encoded_suffix_len]
+                .iter_mut()
+                .for_each(|byte| *byte = rand::random());
```
Comment (kennytm) [inline components/codec/src/byte.rs]: ditto

File: components/codec/src/byte.rs (hunk 1 above)
Comment (breeswish) [reply components/codec/src/byte.rs]: Yes, they are equal, since `payload_encoded`'s length is `encoded_prefix_len + encoded_len + encoded_suffix_len`.

Comment (sticnarf) [reply components/codec/src/byte.rs]: But the upper bound of the range in the original code is `encoded_suffix_len`... Isn't it a bug?

Comment (breeswish) [reply components/codec/src/byte.rs]: Yes. Looks like it is a bug..

**Pull request merged**